hs_core/models.py

```diff
@@ -556,6 +563,16 @@ def create(cls, **kwargs):
             if not kwargs['type'] in dict(cls.SOURCE_TYPES).keys():
                 raise ValidationError('Invalid relation type:%s' % kwargs['type'])
 
+            # ensure isHostedBy and isCopiedFrom are mutually exclusive
+            metadata_obj = kwargs['content_object']
+            metadata_type = ContentType.objects.get_for_model(metadata_obj)
+            if kwargs['type'] == 'isHostedBy' and \
+               Relation.objects.filter(type='isCopiedFrom', object_id=metadata_obj.id, content_type=metadata_type).count() > 0:
```

Comment:
Good suggestion. Replaced count() with exists() in all relevant places and pushed the code change.